GVFS/GVFS.UnitTests.Windows/Windows/Platform/ProjFSFilterTests.cs

```diff
@@ -0,0 +1,185 @@
+﻿using GVFS.Common;
+using GVFS.Common.FileSystem;
+using GVFS.Platform.Windows;
+using GVFS.Tests.Should;
+using GVFS.UnitTests.Category;
+using GVFS.UnitTests.Mock.Common;
+using Moq;
+using NUnit.Framework;
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+
+namespace GVFS.UnitTests.Windows.Platform
+{
+    [TestFixture]
+    public class ProjFSFilterTests
+    {
+        private const string System32DriversRoot = @"%SystemRoot%\System32\drivers";
+        private const string PrjFltDriverName = "prjflt.sys";
+        private const string ProjFSNativeLibFileName = "ProjectedFSLib.dll";
+
+        private readonly string system32NativeLibPath = Path.Combine(Environment.SystemDirectory, ProjFSNativeLibFileName);
+        private readonly string nonInboxNativeLibInstallPath = Path.Combine(ProcessHelper.GetCurrentProcessLocation(), ProjFSNativeLibFileName);
+        private readonly string packagedNativeLibPath = Path.Combine(ProcessHelper.GetCurrentProcessLocation(), "ProjFS", ProjFSNativeLibFileName);
+
+        private readonly string packagedDriverPath = Path.Combine(ProcessHelper.GetCurrentProcessLocation(), "Filter", PrjFltDriverName);
+        private readonly string system32DriverPath = Path.Combine(Environment.ExpandEnvironmentVariables(System32DriversRoot), PrjFltDriverName);
+
+        // .NET doesn't allow us to create custom FileVersionInfos, and so use the version for our assembly and mock
+        // the version comparison methods
+        private readonly FileVersionInfo dummyVersionInfo = FileVersionInfo.GetVersionInfo(Assembly.GetExecutingAssembly().Location);
+
+        private Mock<PhysicalFileSystem> mockFileSystem;
+
+        [SetUp]
+        public void Setup()
+        {
+            this.mockFileSystem = new Mock<PhysicalFileSystem>(MockBehavior.Strict);
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            this.mockFileSystem.VerifyAll();
+        }
+
+        [TestCase]
+        public void IsNativeLibInstalled_ReturnsTrueWhenLibInSystem32()
+        {
+            this.mockFileSystem.Setup(fileSystem => fileSystem.FileExists(this.system32NativeLibPath)).Returns(true);
+            this.mockFileSystem.Setup(fileSystem => fileSystem.FileExists(this.nonInboxNativeLibInstallPath)).Returns(false);
+            ProjFSFilter.IsNativeLibInstalled(new MockTracer(), this.mockFileSystem.Object).ShouldBeTrue();

```

Comment:
As every test includes a new `MockTracer`, you might consider setting this up in the `SetUp` method.